OpenRA.Mods.Common/Traits/World/Selection.cs

```diff
@@ -15,19 +15,28 @@
 using OpenRA.Primitives;
 using OpenRA.Traits;
 
-namespace OpenRA
+namespace OpenRA.Mods.Common.Traits
 {
-	public class Selection
-	{
+    public class SelectionInfo : ITraitInfo
+    {
+		public object Create(ActorInitializer init) { return new Selection(this); }
+    }
+
+    public class Selection : ISelection
+    {
 		public int Hash { get; private set; }
 		public IEnumerable<Actor> Actors { get { return actors; } }
 
 		readonly HashSet<Actor> actors = new HashSet<Actor>();
-		readonly INotifySelection[] worldNotifySelection;
+		INotifySelection[] worldNotifySelection = new INotifySelection[0];
+
+		public Selection(SelectionInfo info)
+		{
+		}
```

Comment:
We usually move empty brackets behind the ctor: `public Selection(SelectionInfo info) { }`